### PR TITLE
Formalize Asset and Coin Terminology in Sway Apps

### PR DESCRIPTION
**Pull request by bitzoic**
## Type of change

- Improvement (refactoring, restructuring repository, cleaning tech debt, ...)

## Changes

The following changes have been made:

- This PR is a continuation of https://github.com/FuelLabs/sway/pull/5461 where the terms "Asset" and "Coin" have been defined and formalized in the Sway lang

### CODE REVIEW

File: DAO/project/SPECIFICATION.md [hunk 1]
```diff
@@ -14,7 +14,7 @@ Table of Content
     - [`user_balance()`](#user_balance)
     - [`user_votes()`](#user_votes)
     - [`proposal()`](#proposal)
-    - [`governance_token_id()`](#governance_token_id)
+    - [`governance_asser_id()`](#governance_asser_id)
```
Comment (SwayStar123) [inline DAO/project/SPECIFICATION.md:17]:
```suggestion
    - [`governance_asset_id()`](#governance_asset_id)
```

File: DAO/project/SPECIFICATION.md [hunk 2]
```diff
@@ -98,9 +98,9 @@ If you are interested in a functional overview then this is the section for you.
    5. The raw number of "yes" and "no" votes
    6. Arbitrary data required for the proposal to execute
 
-### `governance_token_id()`
+### `governance_asser_id()`
```
Comment (SwayStar123) [inline DAO/project/SPECIFICATION.md:101]:
```suggestion
### `governance_asset_id()`
```

File: DAO/project/SPECIFICATION.md (hunk 1 above)
Comment (SwayStar123) [inline DAO/project/SPECIFICATION.md:17]:
```suggestion
    - [`governance_asset_id()`](#governance_asset_id)
```

**Review by SwayStar123 — CHANGES_REQUESTED** [review]

File: airdrop/project/contracts/distributor-contract/src/interface.sw [hunk 3]
```diff
@@ -113,14 +113,3 @@ abi Info {
...
-
-abi SimpleAsset {
-    /// Mints the given amount of tokens to the given Identity.
-    ///
-    /// # Arguments
-    ///
-    /// * `amount`: [u64] - The quantity of tokens to mint.
-    /// * `to`: [Identity] - The user which will receive the minted tokens.
-    #[storage(read, write)]
-    fn mint_to(amount: u64, to: Identity);
```
Comment (SwayStar123) [inline airdrop/project/contracts/distributor-contract/src/interface.sw:125]: Why is this removed?

File: DAO/project/contracts/DAO-contract/tests/functions/info/governance_asset_id.rs [hunk 4]
```diff
@@ -0,0 +1,27 @@
+use crate::utils::{interface::info::governance_asset_id, setup::setup};
+
+mod success {
+    use super::*;
+    use crate::utils::interface::core::constructor;
+
+    #[tokio::test]
+    pub async fn user_can_get_governance_asset_id() {
```
Comment (SwayStar123) [inline DAO/project/contracts/DAO-contract/tests/functions/info/governance_asset_id.rs:8]: Whats with these new tests?

File: airdrop/project/contracts/distributor-contract/src/interface.sw (hunk 3 above)
Comment (bitzoic) [reply airdrop/project/contracts/distributor-contract/src/interface.sw:125]: This is legacy code left over from a previous test suite. Should never have been here

File: DAO/project/contracts/DAO-contract/tests/functions/info/governance_asset_id.rs (hunk 4 above)
Comment (bitzoic) [reply DAO/project/contracts/DAO-contract/tests/functions/info/governance_asset_id.rs:8]: Nothing, it is just renamed to remove the word “token”

**Review by SwayStar123 — APPROVED** [review]

**Review by SwayStar123 — APPROVED** [review]
